### DISCUSSION TITLE
Quartz.net service restarts after starting a job

### DISCUSSION THREAD

**Discussion by gitterdonewell** [quartznet/quartznet#2127] [Q&A]
Quartz.net 3.6.2 (Quartz.Extensions.Hosting) on a .net 6 console app.  As soon quartz starts the cron job, the service restarts without throwing any exception at all.  It runs fine locally on my development laptop but that same job restarts the entire service exactly after 10 seconds in the QA environment.  The console app service runs inside a docker container in kubernate.  Here is what my job looks like:

```
[DisallowConcurrentExecution]
public class SomeJob : IJob
{
     private readonly IDependencyA _dependencyA;
     private readonly IDependencyB _dependencyB;
     private readonly ILogger<SomeJob> _logger;
         
     public SomeJob(IDependencyA dependencyA, IDependencyB dependencyB, ILogger<SomeJob> logger)
    {
        _dependencyA = dependencyA;
        _dependencyB = dependencyB;
        _logger = logger;
    }

    public async Task Execute(IJobExecutionContext contex)
    {   
        try
        {
            _logger.Information("Starting SomeJob...");
            await _dependencyA.DoSomethingAsync(context.CancellationToken);
            await _dependencyB.DoSomethingAsync(context.CancellationToken);
        }
        catch(Exception ex)
        {
            _logger.Error(ex, "Something happened. Unable to run the SomeJob.");
        }
   }    
}
```
program.cs (.net 6)
```
.
.
Log.Information("Starting QuartzNET Console Service....");
.
.
```

When the job starts I see this in the log:

10:00:00 Starting SomeJob...
10:00:10 Starting QuartzNET Console Service...
10:10:00 Application started. Press Ctrl+C to shutdown.  
11:00:00 Starting SomeJob...
11:00:10 Starting QuartzNET Console Service...
11:10:00 Application started. Press Ctrl+C to shutdown.
12:00:00 Starting SomeJob...
12:00:10 Starting QuartzNET Console Service...
12:10:00 Application started. Press Ctrl+C to shutdown.
.
.

So Quartz calls the execute method and I think it crashes calling the _dependencyA.DoSomething() method but no exception is logged even though I'm handling it.  what could be causing this issue?  Thanks

**Comment by jafin** [comment]
Perhaps your job triggers pressure on your kubernetes node and kubernetes is recycling the pod?
I doubt this is a Quartz issue, more so whatever your job is doing may cause  a memory pressure.

Perhaps try running the job context OUTSIDE of quartz to see if problem still occurs. Monitor memory..  Your kubernetes logs may also provide insight why the pod restarts.

> A restarting container can indicate problems with memory (see the Out of Memory section), cpu usage, or just an application exiting prematurely.

**Reply by gitterdonewell** [reply]
I'm going to reach out to the Kubernate team.  
However, as part of troubleshooting, I commented out everything inside the IJob.Execute() method except the _logger.Information() and _logger.Error().  The restart of the console app still happens but this time it randomly occurs.

**Reply by jafin** [reply]
If you still think its quartz, just don't run any jobs, and see if app crashes, thus ruling it out.  
From your description above, the app's pod just looks like its restarting..  I'd see if you can get diagnostics from kubernetes logs to see if it can provide insight why the pod may be  restarting...    You say it runs fine in dev?  Are you able to perhaps monitor memory in dev to see if it sits below whatever threshold the k8 pod has set?  may shed some light if it is memory pressure.

**Reply by gitterdonewell** [reply]
We figured out that the service was getting killed by the sql client and not caused by Quartz.NET
